feat: 생성시간 및 수정시간 관리를 위해 JPA Auditing 적용

Conatuseus opened the pull request:
Squashed commit of the following:

commit 81d1749a6e360c5427ef04a204c422ac8164c350
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:39:47 2019 +0900

    refactor: 값 확인하기 위한 print 추가

commit 884074861bf854037bac29cbc84ac48996288ec7
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:39:31 2019 +0900

    refactor: Order라는 테이블 생성에 문제가 있어 Table 어노테이션 추가 및 이름 부여

commit 8a85f12814a00eca8fba5d8cf739bf06f7185a2f
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:38:52 2019 +0900

    feat: application yml 파일 작성

commit bd75625750746c799e77e343e6d64975938712b8
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:17:41 2019 +0900

    feat: Order test 추가

commit cd21a3baee29a4a7875eec23f6aff1af1f79bb9f
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:17:28 2019 +0900

    feat: Entity들의 createdDate, modifiedDate를 자동으로 관리하는 역할하는 BaseTimeEntity 클래스 추가

commit 848aaec3507c9a328f67aa1b38df2ca4f7f5942f
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:16:53 2019 +0900

    feat: Order를 위한 엔티티, Repository 추가

commit fdf0bc276ed251a8c76fb1ba7b88d83bb38b65aa
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:16:31 2019 +0900

    feat: JPA Auditing 활성화를 위한 어노테이션 추가

Conatuseus committed:
feat: 생성시간 및 수정시간 관리를 위해 JPA Auditing 적용

Squashed commit of the following:

commit 81d1749a6e360c5427ef04a204c422ac8164c350
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:39:47 2019 +0900

    refactor: 값 확인하기 위한 print 추가

commit 884074861bf854037bac29cbc84ac48996288ec7
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:39:31 2019 +0900

    refactor: Order라는 테이블 생성에 문제가 있어 Table 어노테이션 추가 및 이름 부여

commit 8a85f12814a00eca8fba5d8cf739bf06f7185a2f
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:38:52 2019 +0900

    feat: application yml 파일 작성

commit bd75625750746c799e77e343e6d64975938712b8
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:17:41 2019 +0900

    feat: Order test 추가

commit cd21a3baee29a4a7875eec23f6aff1af1f79bb9f
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:17:28 2019 +0900

    feat: Entity들의 createdDate, modifiedDate를 자동으로 관리하는 역할하는 BaseTimeEntity 클래스 추가

commit 848aaec3507c9a328f67aa1b38df2ca4f7f5942f
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:16:53 2019 +0900

    feat: Order를 위한 엔티티, Repository 추가

commit fdf0bc276ed251a8c76fb1ba7b88d83bb38b65aa
Author: MK S <conatuseus@gmail.com>
Date:   Fri Dec 6 21:16:31 2019 +0900

    feat: JPA Auditing 활성화를 위한 어노테이션 추가

yuyu154 commented on src/main/java/com/woowacourse/caffeine/CaffeineApplication.java:
```diff
@@ -2,7 +2,9 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
 
+@EnableJpaAuditing
```
로컬 환경에서 테스트를 돌려봤는데 Shop 관련 테스트가 통과하지 않았어요.
Shop 엔티티에서 BaseTimeEntity를 상속하거나 다른 방법을 쓰는게 좋을 것 같아요

Conatuseus replied:
음... 왜 ShopDocumentTest가 통과하지 않는지 원인을 좀 찾아봐야겠네요. 
감사합니다.

yuyu154 replied:
(https://stackoverflow.com/questions/51467132/spring-webmvctest-with-enablejpa-annotation)

위 글을 참조해봐도 좋을 것 같습니다.

CaffeineApplication 클래스의 @EnableJapAuditing을 다른 클래스로 분리하면 좋을 것 같습니다. 

아래 예시로 했을 때 통과가 되는 것을 확인했습니다.

```
@Configuration
@EnableJpaAuditing
public class TestConfiguration {
}
```

Laterality commented on src/main/java/com/woowacourse/caffeine/repository/OrderRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
불필요한 어노테이션인 것 같아요~

Conatuseus replied:
수정했습니다~!

Laterality commented on src/test/resources/application.yml:
```diff
@@ -0,0 +1,18 @@
+      hibernate:
+        format_sql: true
+    hibernate:
+      ddl-auto: create
```
파일 끝에 빈 줄을 남길 지도 컨벤션으로 정해보면 좋을 것 같아요!

Conatuseus replied:
yml 파일 끝에 빈 줄을 남기는게 보통 컨벤션인가요?
일단 공백줄 추가 해놨습니다.

Conatuseus committed:
refactor: JpaAuditing 어노테이션 위치 변경 및 불필요한 어노테이션 삭제

MockMvc테스트와 충돌로 인해 JpaAuditing 어노테이션을 Config 클래스로 분리
불필요한 Repository 어노테이션 삭제
yml 파일 마지막 공백줄 추가

Squashed commit of the following:

commit 0a9eb4c9ba3fc1e5d4a235d76239aaa8c62606a4
Author: MK S <conatuseus@gmail.com>
Date:   Mon Dec 9 13:56:19 2019 +0900

    refactor: 공백 한 줄 추가

commit 972a9fabf224c7d70ea617a69fe5b97c3bc6dd0e
Author: MK S <conatuseus@gmail.com>
Date:   Mon Dec 9 13:55:37 2019 +0900

    refactor: 불필요한 어노테이션 삭제

commit c0109b6a52bb64311bf21b5e79b2093275696949
Author: MK S <conatuseus@gmail.com>
Date:   Mon Dec 9 13:55:23 2019 +0900

    refactor: 기존의 MockMvcTest과 충돌로 인해 EnableJpaAuditing 어노테이션 분리

Laterality commented on src/test/java/com/woowacourse/caffeine/domain/OrderTest.java:
```diff
@@ -0,0 +1,36 @@
+        //then
+        Order order = orders.get(0);
+
+        System.out.println(">>>>> createDate = "+order.getCreatedDate() + ", modifiedDate = "+order.getModifiedDate());
```
불필요한 코드인 것 같아요

Conatuseus replied:
확인했습니다.

Conatuseus committed:
Update OrderTest.java
